src/components/Marking.vue

```diff
@@ -0,0 +1,199 @@
+              <h1 class="category name">
+                <span style="font-weight: 600"></span>{{ category.type }}
+              </h1>
+              <p class="category subheading">{{ category.desc }}</p>
+            </div>
+            <div class="mark-field">
+              <input type="text" placeholder="1" maxlength="1" class="marks" />
+            </div>
+          </div>
+        </li>
+      </ul>
+    </div>
+    <button style="width=100px;height=100px;" @click="onSubmit()"></button>
+  </div>
+</template>
+
+<script>
+import Vue from "vue";
+import Blurb from "@/components/Blurb.vue";
+import * as firebase from "firebase/app";
+import "firebase/auth";
+import "firebase/firestore";
+import "firebase/storage";
+import db from "../firebaseinit";
+
+export default Vue.extend({
+  name: "Home",
+  components: {
+    Blurb
+  },
+  props: {},
+  data() {
+    return {
+      teams: Array,
+      colors: [
+        ["#FF9DA0", "#FACFC3"],
+        ["#FF9DA0", "#FACFC3"],
+        ["#FF9DA0", "#FACFC3"],
+        ["#649C9F", "#FACFC3"],
+        ["#649C9F", "#FACFC3"]
+      ],
+      categories: [
+        {
+          type: "Technical",
+          desc: "How technically impressive is the hack?"
+        },
+        {
+          type: "Technical",
+          desc: "Is the project complete?"
+        },
+        {
+          type: "Technical",
+          desc: "Does the hack allow for a good user experience?"
+        },
+        {
+          type: "Social Impact",
+          desc: "Does the hack solve an important/relevant issue in society?"
+        },
+        {
+          type: "Social Impact",
+          desc:
+            "Does the hack have a positive impact for the targeted audience?"
+        }
+      ]
+    };
+  },
+  methods: {
+    onSubmit() {
+      let marks = document.getElementsByClassName("marks");
+      let rubric = {};
+      for (let i = 0; i < this.categories.length; i++) {
+        rubric[this.categories[i].desc] = marks[i].value;
+      }
+      const judgeEmail = firebase.auth().currentUser.email;
+      const category = "A";
+      const project = "test1@test.com";
+      const underscore = "_";
+      let arrayIndex = 0;
+      db.collection("DH6")
+        .doc("hackathon")
+        .collection("projects")
+        .doc(project)
+        .get()
+        .then(doc => {
+          console.log(doc.data());
+          let data = doc.data();
+          data = data["_"]["categories"][category];
+          arrayIndex = data.findIndex(x => x.email == "judge@gmail.com");
+        });
+      console.log(arrayIndex);
+      let queryString =
+        underscore + ".categories." + category + ".[" + arrayIndex + "].rubric";
+      console.log(queryString);
+      let updateNested = db
+        .collection("DH6")
+        .doc("hackathon")
+        .collection("projects")
+        .doc(project)
+        .update({
+          _: {
+            categories: {
+              [category]: {
+                [arrayIndex]: {
```

Comment:
Pretty sure this update isn't right, it changes the original structure.
Category is supposed to contain an array of reviews but now after the update its being replaced with an object where the key is the array index

Reply:
I updated the notation so the structure to download the data and then change it in place and reupload it so it should fix these issues

Reply:
Looks good now!